### PR TITLE
Add small collection of `sub_*()` functions

### PR DESCRIPTION
**Pull request by rich-iannone**
The `sub_*()` group of functions are meant to make simple and common substitutions of data in a **gt** table with not-too-much effort. The functions added are:

- `sub_missing()`
- `sub_zero()`
- `sub_small_vals()`
- `sub_large_vals()`

The `sub_missing()` function will replace `fmt_missing()` (deprecating the latter function).

Fixes https://github.com/rstudio/gt/issues/904
Fixes https://github.com/rstudio/gt/issues/910
Fixes https://github.com/rstudio/gt/issues/911

### CODE REVIEW

**Comment by jcheng5** [comment]
Changes requested:

1. Add `prepend=FALSE` argument to `fmt`. If `TRUE`, formatter will be added to the front of the existing list of formatters instead of the end (last one wins instead of first one wins).
2. Introduce `fmt_replace` that just takes static values. e.g. `fmt_replace(value = list(html = "<b>hi</b>", default = "hi"))`
3. `sub_` functions will call `fmt_replace(prepend=TRUE)` instead of having a separate list

**Comment by jcheng5** [comment]
Note to self, advantages of `sub_small_vals` over `fmt_replace` are:

1. No need to know about `prepend = TRUE/FALSE`
2. Can apply to multiple columns in one call
3. Can easily use fancy context-specific symbols

File: R/substitution.R [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+#' 3-18
+#'
+#' @import rlang
+#' @export
+sub_small_vals <- function(
+    data,
+    columns = everything(),
+    rows = everything(),
+    threshold = 0.01,
+    small_pattern = "<{x}",
```
Comment (jcheng5) [inline R/substitution.R:321]:
```suggestion
    small_pattern = if (sign == "+") "<{x}" else md("<*abs*({x})"),
```

File: R/substitution.R [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+#' @section Function ID:
+#' 3-16
+#'
+#' @import rlang
+#' @export
+sub_missing <- function(
+    data,
+    columns = everything(),
+    rows = everything(),
+    missing_text = "---"
```
Comment (jcheng5) [inline R/substitution.R:54]:
```suggestion
    missing_text = md("---")
```

and replace `context_missing_text` with `process_text`

File: R/substitution.R [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+          process_text(
+            context_small_vals_text(
+              threshold = threshold,
+              small_pattern = small_pattern,
+              sign = sign
+            ),
+            context = "html"
+          )
+            ,
+          NA_character_
```
Comment (jcheng5) [inline R/substitution.R:366]: Extract out a common (inner) function

File: R/substitution.R [hunk 4]
```diff
@@ -0,0 +1,609 @@
...
+    columns = {{ columns }},
+    rows = {{ rows }},
+    prepend = TRUE,
+    fns = list(
+      # Any values of `x` that are above the threshold will be processed
+      # according to the `large_pattern`, the `threshold` value (interacts with
+      # the `large_pattern`, and the sign (changes the default `large_pattern`))
+      html = function(x) {
+        ifelse(
+          is.numeric(x) &
```
Comment (jcheng5) [inline R/substitution.R:543]: This `&` doesn't short-circuit. Let's perform a separate check for `is.numeric(x)` and return `rep_len(NA, length(x))`

File: R/substitution.R (hunk 2 above)
Comment (rich-iannone) [reply R/substitution.R:54]: After testing, found that this cannot be done.

File: R/substitution.R (hunk 1 above)
Comment (rich-iannone) [reply R/substitution.R:321]: I used this minor variation: `if (sign == "+") "<{x}" else md("<*abs*(-{x})")`

File: R/substitution.R (hunk 4 above)
Comment (rich-iannone) [reply R/substitution.R:543]: Now done. Thanks for this!

File: R/substitution.R (hunk 3 above)
Comment (rich-iannone) [reply R/substitution.R:366]: Now done.

File: R/substitution.R [hunk 5]
```diff
@@ -348,67 +375,59 @@ sub_small_vals <- function(
...
-          context_small_vals_text(
-            threshold = threshold,
-            small_pattern = small_pattern,
-            sign = sign
-          ),
-          NA_character_
+
+        if (!is.numeric(x)) {
+          return(rep_len(NA_character_, length(x)))
+        }
```
Comment (jcheng5) [inline R/substitution.R:395]: Move these inside the common function

File: R/substitution.R [hunk 6]
```diff
@@ -527,6 +546,33 @@ sub_large_vals <- function(
...
+  sub_replace_large_vals <- function(
+    x,
+    threshold,
+    large_pattern,
+    sign,
+    context
+  ) {
+    ifelse(
+      !is.na(x) &
+        x != 0 &
```
Comment (jcheng5) [inline R/substitution.R:561]:
```suggestion
```

File: R/substitution.R [hunk 7]
```diff
@@ -539,57 +585,59 @@ sub_large_vals <- function(
...
-              sign = sign,
-              context = "html"
-            ),
-            context = "html"
-          ),
-          NA_character_
+
+        if (!is.numeric(x)) {
+          return(rep_len(NA_character_, length(x)))
+        }
```
Comment (jcheng5) [inline R/substitution.R:591]: Also move

**Review by jcheng5 — APPROVED** [review]

**Pull request merged**